flagscale/train/hetero/parallel_context.py

```diff
@@ -594,13 +594,15 @@ def build_all_inter_mesh_process_groups(self):
             )
 
     def build_global_process_groups(self):
+        """ Build global process groups across all process meshes. The global process groups are used for the communication 
+            between different pipeline stages. Heteregonous process groups except for the default process groups are all here"""
         # build global pipeline process groups
         def _backtrack(mesh_index, prev_rank, path, token = "pp", independent_ep=False):
             group_name = self._process_meshes[0].get_group_name(token, independent_ep=independent_ep)
             if mesh_index == len(self._process_meshes):
                 aggregated_ranks = [rank for ranks in path for rank in ranks]
                 self._all_group_ranks[group_name].append(aggregated_ranks)
-                group = torch.distributed.new_group(aggregated_ranks, timeout=self._timeout)
+                group = torch.distributed.new_group(aggregated_ranks, timeout=self._timeout, use_local_synchronization=True)
```

Comment:
Why do we need to enable `use_local_synchronization`?

Reply:
If not enable `use_local_synchronization `, the program can hang when `new_group`